derp/derp_server.go

```diff
@@ -472,7 +495,15 @@ func (s *Server) accept(nc Conn, brw *bufio.ReadWriter, remoteAddr string, connN
 		c.info = *clientInfo
 	}
 
-	s.registerClient(c)
+	for {
+		ok, d := s.registerClient(c)
+		if ok {
+			break
+		}
+		s.clientsReplaceSleeping.Add(1)
+		time.Sleep(d)
```

Comment:
should it sleep d + (something small) or is the rate limiter immune to scheduler quantization type issues where it wakes up only to discover it needs to sleep a bit longer?

Reply:
I don't think anything else is needed. Never seen other x/time/rate users do that. And if there's a thundering herd, it's of a relatively small number of clients per herd. (and usually it's just one, as two nodes fighting is most common)